src/snowflake/snowpark/session.py

```diff
@@ -492,7 +492,8 @@ def _generate_new_action_id(self) -> int:
     def close(self) -> None:
         """Close this session."""
         if is_in_stored_procedure():
-            raise SnowparkClientExceptionMessages.DONT_CLOSE_SESSION_IN_SP()
+            _logger.warning("Closing a session in a stored procedure is a no-op.")
```

Comment:
Two final comments: 1. can we also remove DONT_CLOSE_SESSION_IN_SP() now that it is not used anywhere.

2. Update CHANGELOG.md. Add in improvement section says closing session in SP is a no-op and logs a warning.

Reply:
Sure